src/neptune/attributes/series/fetchable_series.py

```diff
@@ -57,16 +61,40 @@ def make_row(entry: Row) -> Dict[str, Union[str, float, datetime]]:
                 row["timestamp"] = datetime.fromtimestamp(entry.timestampMillis / 1000)
             return row
 
-        progress_bar = False if len(data) < limit else progress_bar
+        if offset is None:
+            offset = 0
+
+        if limit is not None:
+            fetch_chunk_size = min(limit, MAX_FETCH_LIMIT)
+        else:
+            fetch_chunk_size = MAX_FETCH_LIMIT
```

Comment:
Perhaps we could make it into one-liners? Also, I remember creating this `NoLimit` thing that could simplify the code, so that we don't have to check for `limit` being `None` each time